### ISSUE TITLE
[Feature Request] Streamlined bug reporting

### ISSUE THREAD

**Issue by fstirlitz** [yt-dlp/yt-dlp#620]
I don’t like this project’s issue templates.  They are full of obnoxious boilerplate, which is annoying to scroll through when browsing reports and makes them less readable (it sometimes takes me half a minute to realise an issue report contains nothing but a blank template).  They abuse GitHub’s checkboxes feature (meant for progress tracking) for bug reporting checklists.  And the worst part is that they aren’t even particularly effective at what they are supposed to do.  People routinely screw up bug reports by:
- ignoring the template entirely
- leaving out the console output
- posting only part of the console output
- posting *screenshots* of the console output
- forgetting to use the `--verbose` option
- leaving out even *which site* they have a problem with
- leaving in placeholder text like ‘WRITE DESCRIPTION HERE’
- putting the `x` character in brackets *next to* the space instead of replacing the space, breaking checkbox rendering
- using Unicode checkmarks ‘✔’ instead of an ASCII `x`, breaking checkbox rendering

And as people are flocking in here from upstream, which is apparently completely dead now (they haven’t had a commit for a month), this is only going to get worse.  So I think this should be addressed promptly to avoid lots of wasted time, both on the maintainers’ part and reporters’.

So my idea is to add a `--bug-report` option that:
- automatically checks if yt-dlp is at the latest version
- automatically turns on `--verbose` (or even an extra-verbose mode that also lists all available formats)
- captures all console output into a variable
- uses `http.server` to serve up a simple HTML page which contains all captured information so far and a questionnaire for the user
- uses `webbrowser` to open this HTML page in the user’s browser, where the report is **validated** and then posted to GitHub

(Alternatively, we could use Tkinter instead opening a browser, but we would still have to open a web browser eventually to allow the user to log in and submit the report to GitHub.)

A pre-formatted bug report can be sent to GitHub in a GET parameter, e.g. <https://github.com/yt-dlp/yt-dlp/issues/new?title=foo&body=bar>.  This allows about 8K of data to fit in.  We could try to find a workaround if that proves too tight; in the worst case, we could copy the markup into the clipboard and tell the user to paste it into the issue report form.

The resulting report would have all the boilerplate hidden inside a `<details>` tag, to make it less obstructive.  We could then direct people to use the `--bug-report` option to report bugs instead of opening GitHub directly and messing with issue templates.

We could also have a GitHub Action script that checks if issue reports have all the necessary boilerplate in, and *promptly* asks the user to fill in missing information.

**Comment by Ashish0804** [comment]
`--bug-report`
Okay so what do we do when people are not following this and pasting screenshots like always?

`A pre-formatted bug report can be sent to GitHub in a GET parameter`
Umm we will shit tons of DRM issues and network issues.

`We could also have a GitHub Action script that checks if issue reports have all the necessary boilerplate in, and promptly asks the user to fill in missing information.`
Idk how cheap and fast it's going to be but its the best option.
Considering many people don't reply after opening an issue, we could outright close it aswell and let genuine users open another  issue with proper info.

**Comment by fstirlitz** [comment]
>  Okay so what do we do when people are not following this and pasting screenshots like always?

We tell them ‘please re-file using `--bug-report` or follow the bug reporting checklist over here’. Right now we only have the checklist, with umpteenth steps the user has to follow and can therefore screw up. A bug reporting mode within the program can reduce it to just four: enable the mode, fill out the questionnaire, log into GitHub, submit to the tracker.

> Umm we will shit tons of DRM issues and network issues.

When a DRM-encumbered stream is detected, the reporting form can display a relevant warning — and more importantly, *not* display warnings that are irrelevant to the situation at hand and that the user is just going to glaze over thinking ‘TL;DR’. Network issues can be detected similarly: it’s a rather specific kind of error (timeout, connection reset by peer, etc.). The bug report form can suggest to only file retry later.

Users who want to spam the issue tracker with spurious reports can already do so — while skipping the template and therefore not even conveying that the issue is out of scope for this project!

I would be against installing a GitHub Actions hook alone, because I expect that is just going to frustrate users, especially when the Actions hook fails to parse legitimately filled-out fields (compare how `[x]` tends to be screwed up). We should direct users to file informative bug reports in the first place, not have a robot play whack-a-mole by closing incomplete ones. We do that by making the correct action less burdensome: users have only so much energy and attention to spare.

**Comment by fstirlitz** [comment]
Might be helpful: <https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms>

**Comment by pukkandan** [comment]
I guess someone had the same idea for upstream https://github.com/ytdl-org/youtube-dl/issues/29821

**Comment by pukkandan** [comment]
The original suggestion of a fully automated bug reporting is a bit too much imo and I am not sure it would be worth the work that will need to be put into implementing and maintaining it.

On the other hand, replacing our issue templates with forms is a great idea. It is much easier to implement, will somewhat simplify the end user experience and we get some amount of validation for free

**Comment by pukkandan** [comment]
The issue templates have noticeably reduced the amount of invalid issues we get 👍 

Closing this since as mentioned above, I think a fully automated bug reporting is overkill and out of scope

**Comment by Grub4K** [comment]
> - automatically checks if yt-dlp is at the latest version

I agree with this. Maybe we could check known issues and direct people to use the daily builds (or nighty once exists) if the error is listed there as solved, and if it is a common error list the way to solve it directly in the commandline.

> - automatically turns on --verbose (or even an extra-verbose mode that also lists all available formats)
> - captures all console output into a variable

I think this is best implemented after #5680. We can easily employ a custom `Logger` class which can capture the output and analyze it later. This logger can also filter out the bug report message and write out better messages instead. If we decide on what it should do its almost trivial to implement.

Automatically posting the report is something that I would not be doing ([Reference](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-a-url-query)) however automatically preparing the report would work quite well. Spinning up a simple http server might be overkill. For ease of implementation we could copy it to the clipboard instead, and render information after execution in terminal. Then, if that works, go from there ot expand it later on.
